File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

float count_letters(string text);
float count_words(string text);
float count_sentences(string text);

int main(void)
{
    //prompt the user to input text
    string text = get_string("Text: ");

    //call count_letters function
    float letters = count_letters(text);
    //print out the number of letters in the text

    //call count_words function
    float words = count_words(text);
    //print out the number of words in the text

    //call count_sentences function
    float sentences = count_sentences(text);
    //print out the number of sentences in the text

    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;
    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    float grade = roundf(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.f\n", grade);
    }

}

float count_letters(string text)
{
    //isalpha checks if uppercase OR if lowercase
    float sum = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            sum++;
        }
    }
    return sum;
}

float count_words(string text)
{
    //isgraph checks for any characters except space.
    float sum = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isgraph(text[i]) == 0)
        {
            sum++;
        }
    }
    return sum;
}

float count_sentences(string text)
{
    float sum = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sum++;
        }
    }
    return sum;
}

File: pset6/readability.py
from cs50 import get_string


#prompt the user to input text
def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = (int(letters) / int(words)) * 100
    S = (int(sentences) / int(words)) * 100
    index = (0.0588 * L) - (0.296 * S) - 15.8
    grade = round(index)

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    letter_sum = 0
    for a in text:
        if (a.isalpha()) == True:
            letter_sum+=1
    return letter_sum


def count_words(text):
    for b in text:
       word_count = len(text.split())
    word_sum = str(word_count)
    return word_sum


def count_sentences(text):
    sentence_sum = 0
    for c in text:
        if (c == "." or c == "?" or c == "!"):
            sentence_sum += 1
    return sentence_sum


main()
